mage_ai/orchestration/pipeline_scheduler.py

```diff
@@ -472,32 +486,156 @@ def __schedule_blocks(self, block_runs: List[BlockRun] = None) -> None:
+            variables = self.pipeline_run.get_variables(extra_variables={
+                'pipeline.name': self.pipeline.name,
+                'pipeline.uuid': self.pipeline.uuid,
+                'pipeline_uuid': self.pipeline.uuid,
+            })
+            integration_pipeline = IntegrationPipeline.get(self.pipeline.uuid)
+
             self.logger.info(
-                f'Start a process for PipelineRun {self.pipeline_run.id}',
-                **self.__build_tags(),
+                f'Start executing PipelineRun {self.pipeline_run.id}: '
+                f'pipeline {integration_pipeline.uuid}',
+                **tags,
+            )
+
+            executable_block_runs = [b.id for b in block_runs_to_schedule]
+
+            pipeline_schedule = self.pipeline_run.pipeline_schedule
+            schedule_interval = pipeline_schedule.schedule_interval
+            if ScheduleType.API == pipeline_schedule.schedule_type:
+                execution_date = datetime.utcnow()
+            else:
+                # This will be none if trigger is API type
+                execution_date = pipeline_schedule.current_execution_date()
+
+            end_date = None
+            start_date = None
+            date_diff = None
+
+            if ScheduleInterval.ONCE == schedule_interval:
+                end_date = variables.get('_end_date')
+                start_date = variables.get('_start_date')
+            elif ScheduleInterval.HOURLY == schedule_interval:
+                date_diff = timedelta(hours=1)
+            elif ScheduleInterval.DAILY == schedule_interval:
+                date_diff = timedelta(days=1)
+            elif ScheduleInterval.WEEKLY == schedule_interval:
+                date_diff = timedelta(weeks=1)
+            elif ScheduleInterval.MONTHLY == schedule_interval:
+                date_diff = relativedelta(months=1)
+
+            if date_diff is not None:
+                end_date = (execution_date).isoformat()
+                start_date = (execution_date - date_diff).isoformat()
+
+            runtime_arguments = dict(
+                _end_date=end_date,
+                _execution_date=execution_date.isoformat(),
+                _execution_partition=self.pipeline_run.execution_partition,
+                _start_date=start_date,
             )
 
+            data_loader_block = integration_pipeline.data_loader
+            data_exporter_block = integration_pipeline.data_exporter
+
+            all_block_runs = BlockRun.query.filter(BlockRun.pipeline_run_id == self.pipeline_run.id)
+            block_runs = list(filter(lambda br: br.id in executable_block_runs, all_block_runs))
+
+            for stream in parallel_streams_to_schedule:
+                tap_stream_id = stream.get('tap_stream_id')
+                block_runs_for_stream = list(filter(
+                    lambda br: tap_stream_id in br.block_uuid,
+                    block_runs_to_schedule,
+                ))
+                for b in block_runs_for_stream:
```

Comment:
A batch update is more efficient here.
```
    BlockRun.batch_update_status(
        [b.id for b in block_runs_for_stream],
        BlockRun.BlockRunStatus.QUEUED,
    )
```

Reply:
Also, I'm not sure whether it's necessary to update the status of the block runs to QUEUED since the block runs in the same stream are executed in sequence.

Reply:
yeah we actually don't need it